sonar-batch/src/main/java/org/sonar/batch/bootstrap/BatchComponents.java

```diff
@@ -66,8 +67,16 @@ private BatchComponents() {
       HtmlReport.class,
       IssuesReportBuilder.class,
       SourceProvider.class,
-      RuleNameProvider.class
-      );
+      RuleNameProvider.class,
+
+      // Tasks
+      Tasks.class,
+      ListTask.DEFINITION,
+      ListTask.class,
+      ScanTask.DEFINITION,
+      ScanTask.class,
+      ViewsTask.DEFINITION,
+      ViewsTask.class);
```

Comment:
hacking the conversation here

pico is not lazy by default and having a quick look, I didn't see anything making the batch's containers lazy. Maybe I missed it.